.github/workflows/upload_app_store.yml

```diff
@@ -1,6 +1,16 @@
 name: Upload IPA to App Store
 on: workflow_dispatch
 jobs:
+  say_something_if_incorrect_branch:
```

Comment:
If after all we want to have this, IMHO it would be better to have build_test_upload needing this job to succeed instead of having:
```
if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/release') == true }}
```
as a requirement for build_test_upload

Reply:
Well, if something is to be saved, it would be the message, mainly.